src/lib/stores/filtered_items_store.ts

```diff
@@ -0,0 +1,30 @@
+import type { Filters } from 'src/types/filters.type';
+import type { Item } from 'src/types/item.type';
+import { writable } from 'svelte/store';
+
+let gItems: Item[];
+let gFilter: Filters;
+let itemDB: Item[];
+export let displayItems = writable<Item[]>(undefined);
+
+export const filterListener = (filter: Filters) => {
+	gFilter = filter;
+	filterItems();
+};
+
+export const itemsListener = (items: Item[]) => {
+	itemDB = items;
+	filterItems();
+};
+const filterItems = () => {
+	gItems = itemDB;
+	if (gFilter.easy == true) {
+		gItems = gItems.filter((item) => item['easy']);
+	}
+	if (gFilter.search != '') {
+		gItems = gItems.filter((item) => item['name'].toLocaleLowerCase().includes(gFilter.search));
+	}
```

Comment:
This entire file seems to poorly fulfills its role and it's the central part of the filter feature, I wonder if there's an elegant way to substitute these loops

Reply:
Not really easy to remove the loops, to be honest. Since our sample size of work (i.e. the number of items) is still small, I think that having them is fine.